trident/ion_balance.py

```diff
@@ -1071,6 +1008,37 @@ def _internal_ion_fraction_field(field, data):
     return data[(ftype, "%s_number_density" % ion)] / data[(ftype, "%s_nuclei_density" % atom)]
 
 
+def _add_field(ds, name, function, units, sampling_type):
+    """
+    Private function for adding fields that wraps the yt add_field function.
+    First it checks to see if field exists and if so, it does not attempt to
+    add it and gives a warning message to user.  This avoids a misleading
+    warning message from yt about using force_override=True, which no longer
+    applies in Trident.
+    """
+    if name in ds.derived_field_list:
+        mylog.warning("Field ('%s', '%s') already exists. Not clobbering." % (name[0], name[1]))
```

Comment:
You can slim down this line and the one in `_alias_field` by doing this instead:
```
mylog.warning("Field %s already exists. Not clobbering." % str(name))
```